pkg/controller/jobframework/integrationmanager.go

```diff
@@ -186,6 +198,24 @@ func ForEachIntegration(f func(name string, cb IntegrationCallbacks) error) erro
 	return manager.forEach(f)
 }
 
+// EnableIntegration marks the integration identified by name as enabled.
+func EnableIntegration(name string) {
+	manager.enableIntegration(name)
+}
+
+// EnableIntegrationsForTest - should be used only in tests
+// Mark the frameworks identified by names and return a revert function.
+func EnableIntegrationsForTest(tb testing.TB, names ...string) func() {
```

Comment:
I'm not a fan of global states, however in this case when we are doing init bases framework registration this is our best option.

Reply:
It can still be done differently. There could be one `DefaultIntegrationManager` that the implementations can register to.

But, you still can pass a different manager to the objects that need it.

Reply:
Sure it could have been done in a different way, but it was done like this a long time ago and reviewed at that time.

Reply:
Yes, but, at that time, we were not modifying it in tests.